src/app/pages/sharing/smb/smb-form/smb-form.component.ts

```diff
@@ -596,6 +603,18 @@ export class SMBFormComponent implements FormConfiguration, OnDestroy {
         }
       }
     });
+
+    entityForm.formGroup.controls['afp'].valueChanges.pipe(debounceTime(100)).subscribe((res) => {
+      if (res) {
```

Comment:
Let's imagine a case when it is already enabled then user toggles the checkbox (unchecked state). Should we display the dialog?

Reply:
We probably should

Reply:
What if the user checks the box, gets the warning. Decides against it and unchecks it. They should get the warning again?

Reply:
I'd imagine something like:
1. initially **Unchecked** -> Click -> Show the warning -> **Confirm** - `checked` or **Cancel** - `unchecked`
2. initially **Checked** -> Click -> Show the warning -> **Confirm** - `unchecked` or **Cancel** - `checked`

Reply:
Agreed with Denys